app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,6 @@
 class PostsController < ApplicationController
   before_action :set_post, only: %i[show edit update destroy]
-  before_action :set_user, only: %i[index show]
-  before_action :set_current_user, only: %i[index new edit destroy]
+  before_action :set_user, only: %i[show index new create destroy edit update]
```

Comment:
- [  ] You should use `CanCanCan` for Post authorization.
`Hint`: 
Class PostsController < ApplicationController
   load_and_authorize_resource

   ...
end